Core/GDCore/Project/Variable.cpp

```diff
@@ -66,100 +156,123 @@ const Variable& Variable::GetChild(const gd::String& name) const {
   auto it = children.find(name);
   if (it != children.end()) return *it->second;
 
-  isStructure = true;
+  type = Type::Structure;
   children[name] = std::make_shared<gd::Variable>();
   return *children[name];
 }
 
 void Variable::RemoveChild(const gd::String& name) {
-  if (!isStructure) return;
+  if (type != Type::Structure) return;
   children.erase(name);
-  isStructure = !children.empty();
+
+  // If the structure is empty, make it a default empty variable again.

```

Comment:
Same as before, I don't think it's a good idea to change the type and not allow empty structures. It's very surprising and will lead to confusions for sure (and de-optimizations :))